cmd/dockerd/config.go

```diff
@@ -53,6 +55,8 @@ func installCommonConfigFlags(conf *config.Config, flags *pflag.FlagSet) {
 
 	flags.StringVar(&conf.MetricsAddress, "metrics-addr", "", "Set default address and port to serve the metrics api on")
 
+	flags.Var(opts.NewQuotedString(&conf.TrustKeyPath), "key-path", "Path to key file for ID and image signing")
```

Comment:
I think the flag `key-path` might not provide enough context to the user. Could this be `trust-key-path` ?

Reply:
Much prefer it to be `MarkDeprecated()`.  That would have probably saved me two days of head banging and hair pulling trying to find a way around a read-only `/etc/docker` before I discovered this :unamused:

Reply:
@paddy-hack I am currently banging my head - could you clue me in on what you ended up doing to support read-only `/etc`?

It seems like the only solution would be to generate a JWK key manually and point to it with this config option, which seems subpar since dockerd clearly has the ability to generate a key on its own - it just refuses to generate it in any other directory...

Reply:
The "trustkey" has been removed in Docker 23.0 and up; support for pushing the legacy images was removed already (the only remaining purpose for libtrust), and Docker 23.0 now stores an "engine-id" file in `/var/lib/docker/engine-id` (when using the default data location `/var/lib/docker/`), and the daemon will no longer write a file to `/etc/docker/`.

```bash
ls -l /var/lib/docker
total 88
drwx------  2 root root  4096 May 25  2019 builder
drwx------  5 root root  4096 Mar 18  2021 buildkit
drwx--x---  3 root root  4096 Apr  4 20:45 containers
-rw-------  1 root root    59 Mar  7 12:15 engine-id
drwx------  3 root root  4096 Jun 27  2022 image
drwxr-x---  3 root root  4096 May 25  2019 network
drwx--x--- 81 root root 36864 Apr  4 20:49 overlay2
drwx------  4 root root  4096 May 25  2019 plugins
drwx------  2 root root  4096 Mar 28 12:19 runtimes
drwx------  5 root root  4096 Mar 28 12:19 swarm
drwx------  2 root root  4096 Apr  4 20:50 tmp
drwx------  2 root root  4096 May 25  2019 trust
drwx-----x  5 root root  4096 Mar 28 12:19 volumes
```

For existing installations that have a `/etc/docker/key.json` and are upgraded to 23.0, the key ID is copied to the `engine-id` file (to keep the same "ID" in `docker info` output). For fresh installs, or situations where no `key.json` is present, a random ID (GUID) is generated, and used as ID.

For details, see these PRs;

- https://github.com/moby/moby/pull/44566
- https://github.com/moby/moby/pull/44329
- https://github.com/moby/moby/pull/43555

Reply:
@harrisonmg In my case, I was trying to mount a single `docker-daemon.json` "file" from a Kubernetes ConfigMap on `/etc/docker/` in any container our GitLab Runner would spin up for CI/CD jobs.  Doing so resulted in `/etc/docker/` becoming read-only so the daemon could no longer create a `key.json` file at its default location.  I worked around the issue by setting the `deprecated-key-path` in my `docker-daemon.json` file :smile_cat: 

Like so
``` json
{
  "deprecated-key-path": "/var/lib/docker/key.json",
  "registry-mirrors": [
    "https://docker-registry.mirror.internal"
  ]
}
```

@thaJeztah Unfortunately for me, I don't control the Docker versions that many of our CI/CD jobs want to use :crying_cat_face: